### ISSUE TITLE
`SceneNode` and `RootNode` not actually exported from `scenegraph`

### ISSUE THREAD

**Issue by pklaschka** [AdobeXD/typings#56]
From the looks of it, `SceneNode` and `RootNode` aren't actually exported classes in the `scenegraph` module. This is wrongly declared in the typings (`scenegraph.d.ts`).

**Comment by ericdrobinson** [comment]
> This is wrongly declared in the typings (`scenegraph.d.ts`).

Technically, it _appears_ as though it is _correctly_ declared in `scenegraph.d.ts`. We may be looking at a bug in TypeScript. I opened [an issue](https://github.com/microsoft/TypeScript/issues/32182) for clarification or resolution.

In the meanwhile, there are two known ways to work around this:

1. Use interfaces to declare `SceneNode` and its subclasses. This involves defining each `SubClass` as follows:
   ```ts
   interface SubClass extends SceneNode {}
   interface SubClassConstructor {
       new (): SubClass;
   }
   declare const SubClass: SubClassConstructor;
   ```
   This, of course, adds a lot of extra work.
1. Revert to the older version of exporting a specific object (c0d32983507cf4b1909a12e9af1dd6278121ff95):
   ```ts
   export {
       List,
       Each,
       Class,
       Individually,
       ...
   }
   ```

While the second option above will work, it's less desirable as VSCode (and others?) [aren't able to interpret the completions](https://github.com/AdobeXD/typings/issues/54#issuecomment-506974149) as well.

**Comment by ericdrobinson** [comment]
@pklaschka With some insight from a response to [the issue](https://github.com/microsoft/TypeScript/issues/32182#issuecomment-507006868) I posted on the TypeScript repo, I found this method of achieving the desired result:

1. Add `export {};` to the declaration file.
1. Remove the `export` keyword from any class we don't want exported (read: `SceneNode` and `RootNode`).
1. **Rename** `SceneNode` to `SceneNodeClass` (including in `extends SceneNode` _within the file only!_).
    - Do this for `RootNode` as well.
1. Add an interface that extends the class in question: `export interface SceneNode extends SceneNodeClass {}`.
    - Same for `RootNode`.

Taken together it looks something like this:

```ts
export interface SceneNode extends SceneNodeClass {}
declare abstract class SceneNodeClass { ... }

//...

export interface RootNode extends RootNodeClass {}
declare class RootNodeClass extends SceneNodeClass { ... }

//...

export {};
```

_[**Note:** I declared the interface above the class for proximity. It is far easier to see how they are linked this way (no scrolling necessary).]_

This produces roughly the desired effect:

**No `SceneNode` or `RootNode` in value suggestions**
![image](https://user-images.githubusercontent.com/9142587/60397427-33621900-9b1b-11e9-8053-8da36bc13e56.png)

**`SceneNode` and `RootNode` exist in type suggestions**
![image](https://user-images.githubusercontent.com/9142587/60397448-63112100-9b1b-11e9-8382-9e2af307f3a3.png)

---

The one caveat here is that defining the exported types of those classes as `interface` declarations means that types in other module cannot `extend` them (a `class` cannot `extend` an `interface`). If this becomes desirable, the workaround would probably be to create an internal hidden declaration file (named something like `__internal_shared__.d.ts`), define the shared types there, and only _import_ its types into the declarations that use them.

In the above "hidden declaration file" example, you might still wish to export the `type` of the imported class and can do so with the same setup as above (e.g. `export interface BaseNode extends BaseNodeClass {}`, where `BaseNodeClass` had been imported).
